Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/MatchAndMergeWorkspaces.py

```diff
@@ -145,6 +147,21 @@ def fit_x_lims_to_match_histogram_bins(ws_conjoined, x_min, x_max):
             raise AttributeError("Limits contains an infinite value.")
         return x_min, x_max, bin_width
 
+    @staticmethod
+    def exclude_banks(flat_list, x_min, x_max):
+        index_to_remove = []
+        for i in range(x_min.size):
+            if x_min[i] == -1:
+                if x_max[i] == -1:
```

Comment:
This approach of checking x_min before checking x_max doesn't catch a situation where there's an extra -1 in the x_max list. I think you probably need to check `if x_min[i] == -1 and x_max[i]==-1` and then in the else you can check if either is true and raise the error. I happened to do a test with these qlims when I spotted this:
```
q_lim = [[2.5, -1, 4, 6, 7], [3.5, -1, -1, 11, 40]]
```